src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -177,4 +211,26 @@ protected final void onShutdown() {
         // Overridden only to make final so that concrete tool implementations don't override
         super.onShutdown();
     }
+
+    /**
+     *  The emit empty loci parameter comes with several pitfalls when used incorrectly.  Here we check and either give
+     *   warnings or errors.
+     */
+    protected void validateEmitEmptyLociParameters() {
+        if (emitEmptyLoci()) {
+            if (getBestAvailableSequenceDictionary() == null) {
+                throw new UserException.MissingReference("Could not create a sequence dictionary nor find a reference.  Therefore, emitting empty loci is impossible and this tool cannot be run.  The easiest fix here is to specify a reference dictionary.");
+            }
+            if (!hasReference() && !hasIntervals()) {
+                logger.warn("****************************************");
+                logger.warn("* Running this tool without a reference nor intervals can yield unexpected results, since it will emit results for loci with no reads.  A sequence dictionary has been found.  The easiest way avoid this message is to specify a reference.");
+                logger.warn("****************************************");
+            }
+            if (!hasIntervals() && hasReference()) {
+                logger.warn("****************************************");
+                logger.warn("* Running this tool without intervals will cause this tool to run on the entire genome, since it will emit results for loci with no reads.");
```

Comment:
I discussed with @lbergelson  ... I figured that most people would be using this to find coverage at certain loci.  And that if someone was doing this over the whole genome, it was probably an accident.  We do not have strong opinions on this, though.

For now, I am leaving it in.